src/Loggers/LogImage.jl

```diff
@@ -116,211 +139,15 @@ function image_summary(name::AbstractString, imgArray::AbstractArray, format::Im
-    end,
-    HWCN => function(imgArray)
-        @assert ndims(imgArray) == 4
-        N = size(imgArray, 4)
-        for n in 1:N-1
-            log_image(logger, name*"/$n", imgArray[:, :, :, n], HWC, step = step)
-        end
-        name = name*"/$N"
-        permutedims(imgArray[:, :, :, N], (3, 1, 2))
-    end,
-    WHCN => function(imgArray)
-        @assert ndims(imgArray) == 4
-        N = size(imgArray, 4)
-        for n in 1:N-1
-            log_image(logger, name*"/$n", imgArray[:, :, :, n], WHC, step = step)
-        end
-        name = name*"/$N"
-        permutedims(imgArray[:, :, :, N], (3, 2, 1))
-    end,
-    CHWN => function(imgArray)
-        @assert ndims(imgArray) == 4
-        N = size(imgArray, 4)
-        for n in 1:N-1
-            log_image(logger, name*"/$n", imgArray[:, :, :, n], CHW, step = step)
-        end
-        name = name*"/$N"
-        imgArray[:, :, :, N]
-    end,
-    CWHN => function(imgArray)
-        @assert ndims(imgArray) == 4
-        N = size(imgArray, 4)
-        for n in 1:N-1
-            log_image(logger, name*"/$n", imgArray[:, :, :, n], CWH, step = step)
-        end
-        name = name*"/$N"
-        permutedims(imgArray[:, :, :, N], (1, 3, 2))
-    end,
-    NHWC => function(imgArray)
-        @assert ndims(imgArray) == 4
-        N = size(imgArray, 1)
-        for n in 1:N-1
-            log_image(logger, name*"/$n", imgArray[n, :, :, :], HWC, step = step)
-        end
-        name = name*"/$N"
-        permutedims(imgArray[N, :, :, :], (3, 1, 2))
-    end,
-    NWHC => function(imgArray)
-        @assert ndims(imgArray) == 4
-        N = size(imgArray, 1)
-        for n in 1:N-1
-            log_image(logger, name*"/$n", imgArray[n, :, :, :], WHC, step = step)
-        end
-        name = name*"/$N"
-        permutedims(imgArray[N, :, :, :], (3, 2, 1))
-    end,
-    NCHW => function(imgArray)
-        @assert ndims(imgArray) == 4
-        N = size(imgArray, 1)
-        for n in 1:N-1
-            log_image(logger, name*"/$n", imgArray[n, :, :, :], CHW, step = step)
-        end
-        name = name*"/$N"
-        imgArray[N, :, :, :]
-    end,
-    NCWH => function(imgArray)
-        @assert ndims(imgArray) == 4
-        N = size(imgArray, 1)
-        for n in 1:N-1
-            log_image(logger, name*"/$n", imgArray[n, :, :, :], CWH, step = step)
-        end
-        name = name*"/$N"
-        permutedims(imgArray[N, :, :, :], (1, 3, 2))
-    end
+    L => img ->  reshape(img, (1, 1, size(img, 1))),
+    CL => img ->  reshape(img, (size(img, 1), 1, size(img, 2))),
+    LC => img ->  reshape(transpose(img), (size(img, 2), 1, size(img, 1))),
+    HW => img ->  reshape(img, (1, size(img, 1), size(img, 2))),
+    WH => img ->  reshape(transpose(img), (1, size(img, 2), size(img, 1))),
+    HWC => img ->  permutedims(img, (3, 1, 2)),
+    WHC => img ->  permutedims(img, (3, 2, 1)),
+    CHW => img ->  img,
+    CWH => img ->  permutedims(img, (1, 3, 2))
```

Comment:
I don't think the dict of functions gives much that a chain of terneries wouldn't
(and the ternaries would be faster)
```
img = 
	format == L   ? reshape(img, (1, 1, size(img, 1))) :
	format == CL  ? reshape(img, (size(img, 1), 1, size(img, 2))) :
	format == LC  ? reshape(transpose(img), (size(img, 2), 1, size(img, 1))) :
	format == HW  ? reshape(img, (1, size(img, 1), size(img, 2))) :
	format == WH  ? reshape(transpose(img), (1, size(img, 2), size(img, 1))), :
	format == HWC ? permutedims(img, (3, 1, 2)) : 
	format == WHC ? permutedims(img, (3, 2, 1)) :
	format == CHW ? img : 
	format == CWH ? permutedims(img, (1, 3, 2)) :
	#== else ==#    img
```

Reply:
I just tested with @test, ternaries are 8x faster